rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java

```diff
@@ -432,6 +435,63 @@ public static void main(String[] args) {
         }
     }
 
+    private static BlockingQueue<KeyPair> keyPairs = new LinkedBlockingQueue<>();
+
+    private static void launchBackgorundKeyPairMaker(final int numberOfWorkers) {
+        final Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                while (!Thread.interrupted()) {
+                    try {
+                        // if we obeserve less than 'numberOfWorkers' elements than we goint to
+                        // add 'numberOfWorkers' more
+                        if (keyPairs.size() < numberOfWorkers) {
+                            RMNodeStarter.generateAndAddKeyPairs(numberOfWorkers);
+                        }
+
+                        // sleep 1 second, ot until notification (from getKeyPair method)
+                        synchronized (keyPairs) {
+                            keyPairs.wait(1000);
+                        }
+                    } catch (InterruptedException | NoSuchAlgorithmException e) {
+                        logger.warn("Something went terribly wrong.", e);
+                        Thread.currentThread().interrupt();
```

Comment:
If the thread is interrupted or dies somehow we need to make sure that we are still able to produce key pairs otherwise no task will ever be launched further.

Maybe restart a background maker in all cases or fall back to a foreground (like before).

Reply:
If the thread is interrupted, it means the JVM is being stopped, no need to create new key pairs in that case. There is no need to catch the InterruptedException here.